xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -63,21 +63,50 @@ viewers.Comments = Class.create({
   },
   /** Enhance the Comments UI with JS behaviors. */
   startup : function () {
-    if ($("commentform")) {
-      this.form = $("commentform").up("form");
-    } else {
-      this.form = undefined;
-    }
     this.loadIDs();
     this.addReplyListener();
-    this.addSubmitListener(this.form);
-    this.addCancelListener();
     this.addEditListener();
-    this.reloadEditor({
-      callback: function () {
-        this.addPreview(this.form);
-      }.bind(this)
-    });
+    // replaces the comment button with the form on the first click.
+    $("OpenCommentForm").observe('click', function (event) {
+      event.stop();
+      this.toggleFormHidden().then(function () {
+        this.reloadEditor({
+          callback: function () {
+            this.getForm();
+            this.addSubmitListener(this.form);
+            this.addCancelListener();
+            this.addPreview(this.form);
+          }.bind(this)
+        });
+      }.bind(this));
+    }.bind(this));
+  },
+  // lazily load the form element
+  getForm: function () {
```

Comment:
Not sure why you changed that: the form is just hidden but it's present in the HTML so you don't really need to get it lazily.

Reply:
If you keep it like it was you just don't need to change the remaining of the code AFAIU.